src/hashtable.c

```diff
@@ -633,6 +629,19 @@ static int resize(hashtable *ht, size_t min_capacity, int *malloc_failed) {
         }
     }
 
+    if (resize_policy == HASHTABLE_RESIZE_FORBID && ht->tables[0]) {
+        /* Refuse to resize if resizing is forbidden and we already have a primary table. */
+        return 0;
+    }
+    if (exp > old_exp && ht->type->resizeAllowed) {
+        /* If we're growing the table, let's check if the resize is allowed (for now just does a maxmemory checks). */
```

Comment:
We should not imply too much what the caller wants to use this for.

```suggestion
        /* If we're growing the table, let's check if the resize is allowed. */
```